js/ui/codeView.js

```diff
@@ -453,6 +456,44 @@ var CodingSession = new Lang.Class({
         return this._toolbox;
     },
 
+    setGrabbed: function(grabbed) {
+        if (this._button)
+            this._button.opacity = grabbed ? 0 : 255;
+
+        if (this.toolbox && this._state == STATE_APP)
+            this.toolbox.opacity = grabbed ? 0 : 255;
+    },
+
+    _attachAppToToolbox: function() {
+        this._appParent = this.app.get_parent();
+        this._appParent.remove_child(this.app);
+        this.toolbox.insert_child_at_index(this.app, 0);
+        this.app.set_position(0, 0);
+        this._appPositionId = this.app.connect('notify::position', (actor) => {
+            const [x, y] = actor.get_position();
+            if (x || y)
```

Comment:
I was trying to understand why you're checking the position here, instead of just calling `set_position` below, and I suspect it's because otherwise this would end up in an infinite call of `set_position`, as we're triggering the notify from the method we use to connect to it.

Thus, I think this is one of those cases where a comment may avoid future problems of "let me remove these lines and just call `set_position(0, 0)` directly".